core/lib/dal/src/blocks_dal.rs

```diff
@@ -724,161 +746,120 @@ impl BlocksDal<'_, '_> {
-            metadata.aux_data_hash.as_bytes(),
-            metadata.pass_through_data_hash.as_bytes(),
-            metadata.meta_parameters_hash.as_bytes(),
-            metadata.state_diffs_compressed,
-            0,
+            tree_data.hash.as_bytes(),
+            tree_data.rollup_last_leaf_index as i64,
+            number.0 as i64,
         )
+        .instrument("save_batch_tree_data")
+        .with_arg("number", &number)
+        .report_latency()
         .execute(self.storage.conn())
         .await?;
+
+        if update_result.rows_affected() == 0 {
+            tracing::debug!("L1 batch #{number}: tree data wasn't updated as it's already present");
+
+            // Batch was already processed. Verify that existing hash matches
+            let matched: i64 = sqlx::query!(
+                r#"
+                SELECT
+                    COUNT(*) AS "count!"
+                FROM
+                    l1_batches
+                WHERE
+                    number = $1
+                    AND hash = $2
+                "#,
+                number.0 as i64,
+                tree_data.hash.as_bytes(),
+            )
+            .instrument("get_matching_batch_hash")
+            .with_arg("number", &number)
+            .report_latency()
+            .fetch_one(self.storage.conn())
+            .await?
+            .count;
+
+            anyhow::ensure!(matched == 1,
+                "Root hash verification failed. Hash for L1 batch #{} does not match the expected value \
+                 (expected root hash: {:?})",
+                number,
+                tree_data.hash,
+            );
+        }
         Ok(())
     }
 
-    pub async fn save_l1_batch_metadata(
+    pub async fn save_l1_batch_commitment_artifacts(
         &mut self,
         number: L1BatchNumber,
-        metadata: &L1BatchMetadata,
-        previous_root_hash: H256,
-        is_pre_boojum: bool,
+        commitment_artifacts: &L1BatchCommitmentArtifacts,
     ) -> anyhow::Result<()> {
         let mut transaction = self.storage.start_transaction().await?;
 
         let update_result = sqlx::query!(
             r#"
             UPDATE l1_batches
             SET
-                hash = $1,
-                merkle_root_hash = $2,
-                compressed_repeated_writes = $3,
-                compressed_initial_writes = $4,
-                l2_l1_compressed_messages = $5,
-                l2_l1_merkle_root = $6,
-                zkporter_is_available = $7,
-                parent_hash = $8,
-                rollup_last_leaf_index = $9,
-                pass_through_data_hash = $10,
-                meta_parameters_hash = $11,
-                compressed_state_diffs = $12,
+                commitment = $1,
+                aux_data_hash = $2,
+                pass_through_data_hash = $3,
+                meta_parameters_hash = $4,
+                l2_l1_merkle_root = $5,
+                zkporter_is_available = $6,
```

Comment:
Yes, it can be saved when L1 batch sealed but for some reason it was saved with metadata previously. This field is never used when it's read from DB, so I thought about removing it but decided to do it separately to keep diff smaller. It doesn't matter when it's saved and I plan to remove it completely soon, so if you don't mind I will not change this behaviour in this PR